astropy/units/quantity_helper/helpers.py

```diff
@@ -509,12 +471,13 @@ def helper_clip(f, unit1, unit2, unit3):
 for ufunc in radian_to_dimensionless_ufuncs:
     UFUNC_HELPERS[ufunc] = helper_radian_to_dimensionless
 
-# ufuncs handled as special cases
-UFUNC_HELPERS[np.sqrt] = helper_sqrt
-UFUNC_HELPERS[np.square] = helper_square
-UFUNC_HELPERS[np.reciprocal] = helper_reciprocal
-UFUNC_HELPERS[np.cbrt] = helper_cbrt
-UFUNC_HELPERS[np_umath._ones_like] = helper__ones_like
+power_funcs = {np.reciprocal: -1, np.cbrt: Fraction(1, 3), np.sqrt: 0.5, np.square: 2}
+UFUNC_HELPERS |= {
+    f: lambda f, unit, p=power: ([None], one if unit is None else unit**p)
+    for f, power in power_funcs.items()
+}
```

Comment:
The argument for using a `lambda` is weaker in other cases, but these 5 lines achieve the same functionality that the code on current `main` requires 24 lines for.

Reply:
Maybe the line count is not the best way to evaluate the changes. What might be more important is that this one `lambda` (executed 4 times in the comprehension) replaces 4 different functions defined with `def`. In the other cases one `lambda` replaces one `def`, so the benefits of the change are smaller.

> Though it might be clearer still if instead of a `lambda` it used a function creator (say, `make_power_helper`).

I don't see much point in doing that because the natural way of implementing such a factory function would be
```python
def make_power_helper(...):
    return lambda ...
```

Reply:
To me `lambda`s are not that natural, and indeed I think ruff or something will complain about returning a `lambda`...

Instead, I was thinking of
```
def make_power_helper(...):
    def power_helper(f, unit):
        ...

    return power_helper
```
p.s. `lambda` also have the disadvantage that they are not very introspectable; right now,
```
In [3]: u.quantity_helper.UFUNC_HELPERS[np.cbrt]
Out[3]: <function astropy.units.quantity_helper.helpers.helper_cbrt(f, unit)>
```
but this will become `<lambda>(f, unit, power)`

p.s.  If I were starting from scratch, this might even be the following (though likely slower so worse idea):
```
class PowerHelper:
    def __init__(self, power):
        self.power = power
    def __call__(self, f, unit):
        ...
```

Reply:
In the future, if someone new tries to maintain this and have to update this or that to keep with up with numpy, having just one consistent style is better, even though it is 20 more lines of code.

Reply:
I'm fine with this change. I think code comments would help.
A transpiled `class PowerHelper` might be as fast as a lambda.